Symbols required for AOT are private in libmonosgen-2.0.dylib

### Description

Mono's AOT compiler generates references to numerous functions in lib mono, but those functions are not public in  libmonosgen-2.0.dylib, only libmonosgen-2.0.a.

Example for `mono_array_new_1`:

```shell
$ nm microsoft.netcore.app.runtime.mono.iossimulator-x64/6.0.0-preview.7.21329.1/runtimes/iossimulator-x64/native/libmonosgen-2.0.dylib | grep mono_array_new_1 
00000000001a2b40 t _mono_array_new_1

$ nm microsoft.netcore.app.runtime.mono.iossimulator-x64/6.0.0-preview.7.21329.1/runtimes/iossimulator-x64/native/libmonosgen-2.0.a | grep mono_array_new_1
                 U _mono_array_new_1
00000000000004e0 T _mono_array_new_1
```

The end result is that we can't link dynamically with libmonosgen-2.0.dylib when using the AOT compiler, we have to link with the static libmonosgen-2.0.a library.

Example native link failures:

    Undefined symbols for architecture x86_64:
      "_mini_llvm_init_method", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvm_init_method_intptr_intptr_intptr_intptr in System.Private.CoreLib.dll.o
      "_mini_llvmonly_get_interp_entry", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_get_interp_entry_intptr in System.Private.CoreLib.dll.o
      "_mini_llvmonly_init_delegate", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_init_delegate_object in System.Private.CoreLib.dll.o
      "_mini_llvmonly_init_delegate_virtual", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_init_delegate_virtual_object_object_intptr in System.Private.CoreLib.dll.o
      "_mini_llvmonly_init_vtable_slot", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_init_vtable_slot_intptr_int in System.Private.CoreLib.dll.o
      "_mini_llvmonly_pop_lmf", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_pop_lmf_intptr in System.Private.CoreLib.dll.o
      "_mini_llvmonly_resolve_generic_virtual_call", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_resolve_generic_virtual_call_intptr_int_intptr in System.Private.CoreLib.dll.o
      "_mini_llvmonly_resolve_generic_virtual_iface_call", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_resolve_generic_virtual_iface_call_intptr_int_intptr in System.Private.CoreLib.dll.o
      "_mini_llvmonly_resolve_iface_call_gsharedvt", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_resolve_iface_call_gsharedvt_object_int_intptr_intptr in System.Private.CoreLib.dll.o
      "_mini_llvmonly_resolve_vcall_gsharedvt", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_resolve_vcall_gsharedvt_object_int_intptr_intptr in System.Private.CoreLib.dll.o
      "_mini_llvmonly_throw_aot_failed_exception", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_throw_aot_failed_exception_intptr in System.Private.CoreLib.dll.o
      "_mini_llvmonly_throw_nullref_exception", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mini_llvmonly_throw_nullref_exception in System.Private.CoreLib.dll.o
      "_mono_array_new_1", referenced from:
          wrapper_managed_to_native_object___icall_wrapper_mono_array_new_1_intptr_int in System.Private.CoreLib.dll.o
    [...]

Example binlog: [msbuild.binlog.zip](https://github.com/dotnet/runtime/files/6747120/msbuild.binlog.zip)

--- a/src/tasks/AotCompilerTask/MonoAOTCompiler.props
+++ b/src/tasks/AotCompilerTask/MonoAOTCompiler.props
@@ -5,13 +5,15 @@
     <MonoAOTCompilerDefaultAotArguments Condition="'$(TargetArchitecture)' == 'x64'" Include="mtriple=x86_64-ios" />
     <MonoAOTCompilerDefaultAotArguments Condition="'$(TargetArchitecture)' == 'x86'" Include="mtriple=i386-ios" />
     <MonoAOTCompilerDefaultAotArguments Include="static" />
-    <MonoAOTCompilerDefaultAotArguments Include="direct-icalls" />
     <MonoAOTCompilerDefaultAotArguments Include="dwarfdebug" />
     <MonoAOTCompilerDefaultAotArguments Condition="'$(TargetArchitecture)' == 'arm64' or '$(TargetArchitecture)' == 'arm'" Include="mattr=+crc" /> <!-- enable System.Runtime.Intrinsics.Arm (Crc32 and ArmBase for now) -->
     <!--<MonoAOTCompilerDefaultAotArguments Include="direct-pinvoke" />--> <!-- TODO: enable direct-pinvokes (to get rid of -force_loads)-->
   </ItemGroup>
   <ItemGroup Condition="'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'iOSSimulator' or '$(TargetOS)' == 'tvOS' or '$(TargetOS)' == 'tvOSSimulator' or '$(TargetOS)' == 'MacCatalyst'">
     <MonoAOTCompilerDefaultProcessArguments Include="-O=gsharedvt" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'tvOS' or '$(TargetOS)' == 'MacCatalyst'">
+    <MonoAOTCompilerDefaultAotArguments Include="direct-icalls" />
   </ItemGroup>
   <ItemGroup Condition="'$(TargetOS)' == 'Android'">
     <MonoAOTCompilerDefaultAotArguments Condition="'$(TargetArchitecture)' == 'arm'" Include="mtriple=armv7-linux-gnueabi" />

